differential-dataflow/src/consolidation.rs

```diff
@@ -31,55 +32,61 @@ pub fn consolidate<T: Ord, R: Semigroup>(vec: &mut Vec<(T, R)>) {
 /// This method will sort `vec[offset..]` and then consolidate runs of more than one entry with
 /// identical first elements by accumulating the second elements of the pairs. Should the final
 /// accumulation be zero, the element is discarded.
+#[inline]
 pub fn consolidate_from<T: Ord, R: Semigroup>(vec: &mut Vec<(T, R)>, offset: usize) {
     let length = consolidate_slice(&mut vec[offset..]);
     vec.truncate(offset + length);
 }
 
 /// Sorts and consolidates a slice, returning the valid prefix length.
+#[inline]
 pub fn consolidate_slice<T: Ord, R: Semigroup>(slice: &mut [(T, R)]) -> usize {
-
     if slice.len() > 1 {
+        consolidate_slice_slow(slice)
+    }
+    else {
+        slice.iter().filter(|x| !x.1.is_zero()).count()
```

Comment:
Probably not the most important thing, but I wonder if there is a way to write this that is more helpful to Rust. Like, rather than iter and hope Rust realizes the slices are either len 0 or len 1, we could `get(0)` and `unwrap_or(0)`?

Reply:
Ah, I see these were inherited from the previous implementation! We can explore in follow-up work, or .. ignore if no improvements!